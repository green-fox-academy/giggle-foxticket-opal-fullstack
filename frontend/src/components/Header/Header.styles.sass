.NavbarItems
  background: linear-gradient(to right, #3cb878d9, #14ec7dd9)
  height: 80px
  display: flex
  justify-content: center
  align-items: center
  font-size: 1.2rem

  img
    margin-left: 2rem
    width: 5rem
    justify-self: start

.navbar-logo
  color: white
  justify-self: start
  cursor: pointer

.nav-menu
  display: grid
  grid-template-columns: repeat(4, auto)
  column-gap: 2rem
  list-style: none
  text-align: center
  width: 90vw
  justify-content: end
  margin-right: 2rem

.nav-links
  cursor: pointer
  padding: 10px
  border-radius: 14px
  transition: all 350ms ease-in-out
  color: inherit
  text-decoration: none

  &:hover
    filter: brightness(115%)
    box-shadow: 0px 0px 12px 0.8px #3cb878
    color: #bbffdc

#fox-brand
  color: #bbffdc
  font-size: 2.2rem
  margin-left: 2rem

