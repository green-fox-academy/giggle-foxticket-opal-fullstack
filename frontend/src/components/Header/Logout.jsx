import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { LOGOUT_SUCCESS } from '../../flux/actions/types';

function Logout() {
  const dispatch = useDispatch();
  return (
    <>
      <Link
        to="/"
        className="nav-links mt-1"
        onClick={() => {
          dispatch({ type: LOGOUT_SUCCESS });
        }}
      >
        Logout
      </Link>
    </>
  );
}

Logout.propTypes = {
  logout: PropTypes.func,
};

export default Logout;
