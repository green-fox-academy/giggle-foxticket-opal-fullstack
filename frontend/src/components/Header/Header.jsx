import React from 'react'
import FoxticketLogo from '../../assets/images/logos/Foxticket.png'
import Logout from './Logout';
import { Link } from "react-router-dom";
import { connect } from 'react-redux';
import './Header.styles.sass'

function Header({userName, isAdmin}) {
  return (
      <nav className="NavbarItems" >
        <img src={FoxticketLogo} alt='navigation bar logo'></img>
        <h3 className="nav-text">Foxticket</h3>
        <ul className="nav-menu">
        {
          userName &&
          <>
          <li className="nav-links">{userName}</li>
          <Logout />
          </>
        }
        { 
          isAdmin &&
          <Link to="/admin" className="nav-links" >Admin</Link>
        }
        </ul>
      </nav>
  )
}

const mapStateToProps = (state) => {
   if(state.auth.user) {
      return {
        userName: state.auth.user,
        isAdmin: state.auth.isAdmin
      }
   } else {
      return {
        userName: null,
        isAdmin: false
      }
   }
}

export default connect (mapStateToProps)(Header)
