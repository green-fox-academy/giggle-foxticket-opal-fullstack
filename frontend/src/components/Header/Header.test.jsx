import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('Connect component to Redux store', () => {
  Storage.prototype.getItem = jest.fn(() => 'token');
  const mockStore = configureStore([]);
  const user = {
    name: 'Someone',
    isAdmin: true
  }
  const store = mockStore({ user });

  it('should render with given state from Redux store', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </BrowserRouter>)
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
