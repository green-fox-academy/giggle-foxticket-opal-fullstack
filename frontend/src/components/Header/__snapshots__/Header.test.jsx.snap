// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to Redux store should render with given state from Redux store 1`] = `
<nav
  className="NavbarItems"
>
  <img
    alt="navigation bar logo"
    src="Foxticket.png"
  />
  <h3
    className="nav-text"
  >
    Foxticket
  </h3>
  <ul
    className="nav-menu"
  >
    <a
      className="nav-links"
      href="/admin"
      onClick={[Function]}
    >
      Admin
    </a>
    <li
      className="nav-links"
    >
      Someone
    </li>
    <a
      className="nav-links"
      href="/"
      onClick={[Function]}
    >
      Logout
    </a>
  </ul>
</nav>
`;
